bitutils/bitutils_test.go

```diff
@@ -0,0 +1,149 @@
+package bitutils
+
+import (
+	"testing"
+)
+
+// Define some test data
+
+// 0000 0100 1010 0010 0000 0011 1011 0001 0000 0000 0010 1011
+
+var testdata = []byte{0x04, 0xa2, 0x03, 0xb1, 0x00, 0x2b}
+
+type testDefinition struct {
+	data   []byte // The data to feed the function
+	offset uint   // The bit offset in the byte slice to start
+	value  uint64 // The value we expect the function to return (64 bit to allow for future functions that extract larger ints)
+}
+
+var test4Bits = []testDefinition{
+	{testdata, 21, 7},
+	{testdata, 12, 2},
+	{testdata, 44, 11},
+	{testdata, 6, 2},
+	{[]byte{0x10}, 0, 1},
+	{[]byte{0x92}, 4, 2},
+	{[]byte{0x99}, 1, 3},
+	{[]byte{0x01, 0xe0}, 7, 15},
+}
+
+func TestParseByte4(t *testing.T) {
+	b, err := ParseByte4(testdata, 46)
+	assertStringsEqual(t, "ParseByte4 expected 4 bits to start at bit 46, but the consent string was only 6 bytes long (needs second byte)", err.Error())
+
+	b, err = ParseByte4(testdata, 80)
+	assertStringsEqual(t, "ParseByte4 expected 4 bits to start at bit 80, but the consent string was only 6 bytes long", err.Error())
+
+	for _, test := range test4Bits {
+		b, err = ParseByte4(test.data, test.offset)
+		assertNilError(t, err)
+		assertBytesEqual(t, byte(test.value), b)
+	}
+}
+
+// Used https://cryptii.com/ to convert 8 bit sequeces to integers
+var test8Bits = []testDefinition{
+	{testdata, 4, 0x4a},
+	{testdata, 7, 81},
+	{testdata, 26, 196},
+	{testdata, 6, 40},
+}
```

Comment:
Figured with the list duplicates were better than not, in case one "duplicate" differed in a way that was not anticipated to matter, but actually does.

Reply:
The "duplicates" also have different bit patterns, which is one way there could be a subtle, unexpected difference between "duplicates".